R/ff_predict.R

```diff
@@ -75,35 +129,36 @@ ff_predict <- function(model, test_matrix, threshold = 0.5, groundtruth = NA, in
   } else {
     model_features <- model$feature_names
   }
+  return(model_features)
+}
+
+FeaturePrepper <- function(model, test_matrix, threshold, groundtruth, indices, templateraster, verbose, certainty, mdlftrs) {
   if (!is.null(model_features)) {
     test_features <- colnames(test_matrix$features)
     # Check for features in the test matrix not present in the model
-    extra_features <- setdiff(test_features, model_features)
+    extra_features <- setdiff(test_features, mdlftrs)
     # If there are extra features, remove them from the test matrix
     if (length(extra_features) > 0) {
       ff_cat(
```

Comment:
When the brackets opens with nothing behind it directly, convention is to close with a bracket on a seperate line